### DISCUSSION TITLE
Cant download firmware by cfclient

### DISCUSSION THREAD

**Discussion by SUJUSTDO** [bitcraze/discussions#1634] [Q&A] [answered]
I am trying to use cfclient to flash the firmware of crazyfile (my current version is 2024.10). First I want to flash from release, but the platform filter does not have a file to choose from. So I choose to flash from file, but when I click program to start flashing, cfclient disconnects. I have tried this several times, even replacing the crazyfile with another one.

**Comment by ToveRumar** [comment, marked as answer]
Hi!

Thank you for your question!
I need some more information about your current setup.

- What platform are you using, is it Crazyflie 2.1? 
- What FW version is on it now?
- When you say that the platform filter does not have a file to choose from, what do you mean?  Could you give me some output logs from the client?
- What client version are you using? Could you try with the latest one?

This is how it looks for me when I am looking through the releases for Crazyflie 2.1

![Screenshot from 2024-11-06 16-09-44](https://github.com/user-attachments/assets/df65cabe-7120-47fd-8a8e-25065dfaa58e)


If you are flashing your own zip file, make sure that the contents and naming matches the release zip files.
You can download the release files here, https://github.com/bitcraze/crazyflie-release/releases/tag/2024.10

**Reply by SUJUSTDO** [reply]
Hi, I've encountered the same problem again and I'll give you some specific information.
1. I'm using Crazyfile 2.1.
2. My cflclient is the latest version (2024.11)
3. I can't update the firmware by selecting the platform filter From release. Also, when I try From file, it always disconnects from cfclient.
4. The drone currently has the latest firmware (2024.11).
I also have some other circumstances to explain. I have tried other versions of cfclient and had the same problem. Even if I uninstalled and reinstalled it, the result was the same. How can I solve this?
![1](https://github.com/user-attachments/assets/941bf86e-0c05-49f5-9958-b4039876030a)

**Reply by Bananazooka** [reply]
Hi, 
Could you make a new discussion about this problem? Remember to provide as much information as possible. The [cfclients](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/)'s debug tab (found under 'Help'>'about') can already provide handy information about your development environment and your connected Crazyflie. Copy pasting the[ console tab's content](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/console_tab/) also helps a lot.
